app/views/games/show.html.erb

```diff
@@ -1,9 +1,44 @@
-<h1>Game: <%= @game.id %></h1>
+<script>
+  $(function() {
+    $('.piece').draggable( {
+      containment: .board,
+      cursor: 'crosshair',
+      snap: false,
+      revert: invalidMove,
+    });
+    // function invalidMove(event, ui)
+
+    $('.board').droppable( {
+      tolerance: 'fit',
+      drop: dropConditions,
+      }
+      update: function(event, ui) {
+        $.ajax({
+          type: 'PUT',
+          url: ui.game.data('game-id'),
```

Comment:
Here you want to check the data attributes for the piece that's selected. I'm not sure exact code in jquery.

Here is rough idea...
```
$.ajax({
  type: 'PUT',
  url: ui.draggable.data('update-url'),
  dataType: 'json',
  data: {piece: {x_coordinate: something, y_coordinate: something}}
}).done(function( data ) {
  // was successful, may have to remove another piece from the board here.
});
```